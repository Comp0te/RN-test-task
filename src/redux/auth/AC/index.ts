import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';

export enum ActionTypes {
  AUTH_LOGOUT = 'AUTH_LOGOUT',
  AUTH_SET_IS_AUTH_USER = 'AUTH_SET_IS_AUTH_USER',
}

export const Actions = {
  logout: () => createAction(ActionTypes.AUTH_LOGOUT),
  setIsAuthUser: (data: boolean) => createAction(ActionTypes.AUTH_SET_IS_AUTH_USER, {data}),
};

export type Actions = ActionsUnion<typeof Actions>;
