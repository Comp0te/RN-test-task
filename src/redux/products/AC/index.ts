import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';
import { ProductModel } from '../../../shared/models/product.model';

export enum ActionTypes {
  SET_PRODUCTS_DATA = 'SET_PRODUCTS_DATA',
  SET_SEARCH_QUERY = 'SET_SEARCH_QUERY',
}

export const Actions = {
  setProductsData: (data: ProductModel[]) => createAction(ActionTypes.SET_PRODUCTS_DATA, {data}),
  setSearchQuery: (data: string) => {
    return createAction(ActionTypes.SET_SEARCH_QUERY, {data});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
