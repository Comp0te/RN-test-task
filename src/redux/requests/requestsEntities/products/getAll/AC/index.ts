import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ProductModel } from '../../../../../../shared/models/product.model';

export enum ActionTypes {
  PRODUCTS_GET_ALL_REQUEST = 'PRODUCTS_GET_ALL_REQUEST',
  PRODUCTS_GET_ALL_REQUEST_SUCCESS = 'PRODUCTS_GET_ALL_REQUEST_SUCCESS',
  PRODUCTS_GET_ALL_REQUEST_FAIL = 'PRODUCTS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllProducts: () => createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST),
  getAllProductsSuccess: (data: ProductModel[]) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllProductsFail: (errors: string) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
