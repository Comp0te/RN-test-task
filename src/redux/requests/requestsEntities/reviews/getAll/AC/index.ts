import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ReviewResponse } from '../../../../../../shared/models/review.model';

export enum ActionTypes {
  REVIEWS_GET_ALL_REQUEST = 'REVIEWS_GET_ALL_REQUEST',
  REVIEWS_GET_ALL_REQUEST_SUCCESS = 'REVIEWS_GET_ALL_REQUEST_SUCCESS',
  REVIEWS_GET_ALL_REQUEST_FAIL = 'REVIEWS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllReviews: (data: string) => createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST, {data}),
  getAllReviewsSuccess: (data: ReviewResponse[]) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllReviewsFail: (errors: string) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
