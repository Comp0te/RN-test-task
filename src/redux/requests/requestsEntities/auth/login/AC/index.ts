import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  LOGIN_REQUEST = 'LOGIN_REQUEST',
  LOGIN_REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS',
  LOGIN_REQUEST_FAIL = 'LOGIN_REQUEST_FAIL',
}

export const Actions = {
  login: (data: AuthInput) => createAction(ActionTypes.LOGIN_REQUEST, {data}),
  loginSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.LOGIN_REQUEST_SUCCESS, {data});
  },
  loginFail: (errors: string) => {
    return createAction(ActionTypes.LOGIN_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
