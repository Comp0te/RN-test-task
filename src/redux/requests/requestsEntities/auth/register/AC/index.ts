import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  REGISTER_REQUEST = 'REGISTER_REQUEST',
  REGISTER_REQUEST_SUCCESS = 'REGISTER_REQUEST_SUCCESS',
  REGISTER_REQUEST_FAIL = 'REGISTER_REQUEST_FAIL',
}

export const Actions = {
  register: (data: AuthInput) => createAction(ActionTypes.REGISTER_REQUEST, {data}),
  registerSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.REGISTER_REQUEST_SUCCESS, {data});
  },
  registerFail: (errors: string) => {
    return createAction(ActionTypes.REGISTER_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
