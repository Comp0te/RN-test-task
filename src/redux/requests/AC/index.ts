import * as login from '../requestsEntities/auth/login/AC';
import * as register from '../requestsEntities/auth/register/AC';
import * as getAllProducts from '../requestsEntities/products/getAll/AC';
import * as getAllReviews from '../requestsEntities/reviews/getAll/AC';
import * as postReview from '../requestsEntities/reviews/post/AC';

export const requestsAC = {
  login,
  register,
  getAllProducts,
  getAllReviews,
  postReview,
};