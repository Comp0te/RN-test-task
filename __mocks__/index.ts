import { AuthInput } from '../src/shared/services/auth.service';
import { AuthResponse } from '../src/redux/requests/types';
import { ProductModel } from '../src/shared/models/product.model';
import { ReviewResponse } from '../src/shared/models/review.model';
import { IReviewPostInput } from '../src/shared/services/reviews.service';

export const authInputMock: AuthInput = {
  username: 'Compote',
  password: 'password',
};

export const authResponseMock: AuthResponse = {
  success: true,
  token: 'token',
};

export const authResponseFailMock = {
  success: false,
  message: 'Ooops',
};

export const productMock: ProductModel = {
  id: 1,
  img: 'img',
  text: 'text',
  title: 'title',
};

export const reviewResponseMock: ReviewResponse = {
  id: 1,
  created_by: {
    id: 1,
    username: 'bla-bla',
  },
  product: 1,
  text: 'text',
  rate: 5,
};

export const reviewPostInputMock: IReviewPostInput = {
  text: 'text',
  rate: 2,
  product_id: 'id',
};