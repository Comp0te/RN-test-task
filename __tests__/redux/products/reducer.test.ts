import { Actions, ActionTypes } from '../../../src/redux/products/AC';
import {
  productsReducer,
  initialState as productsInitialState,
} from '../../../src/redux/products/reducers';
import { productMock } from '../../../__mocks__';

describe('Redux products reducers', () => {
  it(`update state after ${ActionTypes.SET_PRODUCTS_DATA} action`, () => {
    expect(productsReducer(
      productsInitialState,
      Actions.setProductsData([productMock, productMock])),
    ).toMatchSnapshot();
  });

  it(`update state after ${ActionTypes.SET_SEARCH_QUERY} action`, () => {
    expect(productsReducer(
      productsInitialState,
      Actions.setSearchQuery('title')),
    ).toMatchSnapshot();
  });
});
