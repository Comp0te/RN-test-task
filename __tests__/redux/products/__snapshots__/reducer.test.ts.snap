// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux products reducers update state after SET_PRODUCTS_DATA action 1`] = `
Object {
  "allIds": Array [
    "1",
  ],
  "entities": Object {
    "1": ProductModel {
      "id": 1,
      "img": "img",
      "text": "text",
      "title": "title",
    },
  },
  "searchQuery": "",
}
`;

exports[`Redux products reducers update state after SET_SEARCH_QUERY action 1`] = `
Object {
  "allIds": Array [],
  "entities": Object {},
  "searchQuery": "title",
}
`;
